src/utils/sharedPromiseTo.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * @template T
+ * @param { () => Promise<T> } [defaultFunc]
+ * @returns { (func?: () => Promise<T>) => Promise<T> }
+ */
+module.exports = defaultFunc => {
```

Comment:
Do we need `defaultFunc`? It's currently not used anywhere. I would also use curly braces and make this a little bit less dense, wdyt?

Reply:
I think you are right - it is trying to be too clever.  I had both options because it's slightly easier to use the 'invocation time' method from a class (well, less code changes are needed to adopt it.)

I have changed to only provide option at initialisation time.  Let me know what you think.